bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValueImpl.java

```diff
@@ -300,7 +300,7 @@ public byte getByte(long index) {
         if (byteValues != null) {
             return byteValues[(int) index];
         }
-        return (Byte) refValues[(int) index];
+        return ((Integer) refValues[(int) index]).byteValue();
```

Comment:
A b7a `byte[]` is represented as a Java `byte[]`.

Everywhere else, a b7a `byte` is represented by a Java `int` (values 0 to 255).

So in this particular scenario, when a b7a `byte` is in a ref value array (which is represented as a Java `Object[]`), it'll be as a Java `Integer`, and we can't directly cast it to a Java `Byte`. 

IINM we wouldn't miss anything with this approach since the compiler guarantees that a `byte` can only have a value in the range `0` to `255`. So this `Integer` will be some value in the range `0` to `255` and there's a direct representation when we do `.byteValue()` on it (-128 to 127).